app/models/residence.rb

```diff
@@ -33,8 +33,12 @@ def initialize_residence
   end
 
   def to_s
-    if current
+    if current && /^homeless/i.match(address.to_s)
+     "Currently homeless"
+    elsif current
       "Current residence at #{address}"
+    elsif /homeless/i.match(address.to_s)
```

Comment:
Good call on this. Plopped a homeless? method in the address model, which returns true if the street given has homeless in the name, and called that in the residence model instead.